storage/innobase/buf/buf0buf.cc

```diff
@@ -921,23 +922,17 @@ void mem_pressure::pressure_routine(mem_pressure *m)
         }
       }
 
+#ifdef UNIV_DEBUG
       if (p.revents & POLLIN)
       {
-        char c= '\0';
-        /* signal to quit */
-        if (read(p.fd, &c, 1) >=0)
-          switch (c) {
-          case 'Q':
-            goto shutdown;
-#ifdef UNIV_DEBUG
-          case 'G':
-            buf_pool.garbage_collect();
-#endif
-          }
+        uint64_t u;
+        /* we haven't aborted, so this must be a debug trigger */
+        if (read(p.fd, &u, sizeof(u)) >=0)
```

Comment:
We had written 8 bytes, but here (depending on the kernel-side implementation) we could theoretically read that in multiple requests, so a single debug event could theoretically trigger multiple garbage collections. Not a big deal.

Reply:
the way its describes seems rather strict on 8 bytes. But who doesn't like a bit of debug fuzzing :-)